app/services/syncrhonize_users_service.rb

```diff
@@ -0,0 +1,18 @@
+class SyncrhonizeUsersService
+  attr_reader :users
+
+  def initialize(users)
+    @users = users
+  end
+
+  def perform
+    @users.each do |member|
+      user = User.find_or_create_by(login: member.login)
```

Comment:
with `find_or_create_by` and then an `update_attributes` you'll have two SQL statements executed to create/update the record, plus one to try and find the record by `login`. Did you try `find_or_initialize_by` to avoid that one extra `INSERT` to the database ?

Reply:
Awesome, had to read the docs to understand what those methods are doing under the water... and also your help was great :)